src/ui/public/kuery/suggestions/index.js

```diff
@@ -0,0 +1,32 @@
+import { flatten } from 'lodash';
+import { fromKueryExpression } from '../ast';
+import * as field from './field';
+import * as value from './value';
+import * as operator from './operator';
+import * as conjunction from './conjunction';
+import * as recentSearch from './recent_search';
+
+const suggestionProviders = { field, value, operator, conjunction, recentSearch };
+const cursor = '\0';
+
+export function getSuggestionsProvider({ $http, config, indexPattern, persistedLog }) {
+  return ({ query, selectionStart, selectionEnd }) => {
+    const cursoredQuery = `${query.substr(0, selectionStart)}${cursor}${query.substr(selectionEnd)}`;
+
+    let cursorNode;
+    try {
+      cursorNode = fromKueryExpression(cursoredQuery, { parseCursor: true });
+    } catch (e) {
+      cursorNode = {};
+    }
+
+    const { suggestionTypes = [] } = cursorNode;
+    return Promise.all([...suggestionTypes, 'recentSearch'].map(type => {
```

Comment:
Putting `recentSearch` first seems more natural. Otherwise, when you focus the query bar, you just get a huge list of fields and need to scroll down (with the mouse) to see recent searches:

![screenshot 2018-03-09 14 01 55](https://user-images.githubusercontent.com/404731/37229879-c2f4de14-23a2-11e8-996e-2cc95d99f3b0.png)

Versus:

![screenshot 2018-03-09 14 08 40](https://user-images.githubusercontent.com/404731/37230094-7bb3f980-23a3-11e8-89f7-039bdc1dd7d1.png)